modules/programs/firefox.nix

```
@@ -42,7 +42,7 @@ let
 
   mkUserJs = prefs: extraPrefs: bookmarks:
     let
-      prefs' = lib.optionalAttrs ({ } != bookmarks) {
+      prefs' = lib.optionalAttrs ([ ] != bookmarks) {
```

Since the `bookmarks` option could also be an attribute set (when using the old structure), this should check that `bookmarks` is not an empty list nor an empty set. The same should probably be done when calling `mkUserJs`.
```suggestion
      prefs' = lib.optionalAttrs (bookmarks != [ ] && bookmarks != { }) {
```

---

I'll change this. Thank you